src/terms/hartree.jl

```diff
@@ -29,9 +29,13 @@ end
 function TermHartree(basis::PlaneWaveBasis{T}, scaling_factor) where {T}
     # Solving the Poisson equation ΔV = -4π ρ in Fourier space
     # is multiplying elementwise by 4π / |G|^2.
-    poisson_green_coeffs = 4T(π) ./ [sum(abs2, G) for G in G_vectors_cart(basis)]
-    poisson_green_coeffs[1] = 0  # Compensating charge background => Zero DC
+
+    GPUArraysCore.allowscalar() do
+        poisson_green_coeffs = 4T(π) ./ [norm2(G) for G in G_vectors_cart(basis)]
+        poisson_green_coeffs[1] = 0  # Compensating charge background => Zero DC
```

Comment:
only enclose this one in allowscalar? also is there no macro that avoids the explicit closure?